docs/src/main/paradox/scala/http/routing-dsl/testkit.md

```diff
@@ -77,17 +80,88 @@ The following inspectors are defined:
+|`assertEntityAs(Unmarshaller<T> unmarshaller, expectedValue: T)`    | Asserts that the entity data if unmarshalled with the given marshaller equals the given value                                       |
+|`assertHeaderExists(HttpHeader expectedHeader)`                     | Asserts that the response contains an HttpHeader instance equal to the expected one                                                 |
+|`assertHeaderKindExists(String expectedHeaderName)`                 | Asserts that the response contains a header with the expected name                                                                  |
+|`assertHeader(String name, String expectedValue)`                   | Asserts that the response contains a header with the given name and value.                                                          |
+
+It's, of course, possible to use any other means of writing assertions by inspecting the properties the response
+manually. As written above, `TestResponse.entity` and `TestResponse.response` return strict versions of the
+entity data.
+
+## Supporting Custom Test Frameworks
+
+Adding support for a custom test framework is achieved by creating new superclass analogous to
+`JUnitRouteTest` for writing tests with the custom test framework deriving from `akka.http.javadsl.testkit.RouteTest`
+and implementing its abstract methods. This will allow users of the test framework to use `testRoute` and
+to write assertions using the assertion methods defined on `TestResponse`.
+
+
+@@@
+
 ## Testing sealed Routes
 
 The section above describes how to test a "regular" branch of your route structure, which reacts to incoming requests
 with HTTP response parts or rejections. Sometimes, however, you will want to verify that your service also translates
 @ref[Rejections](rejections.md) to HTTP responses in the way you expect.
 
-You do this by wrapping your route with the `akka.http.scaladsl.server.Route.seal`. The `seal` wrapper applies the logic of the in-scope @ref[ExceptionHandler](exception-handling.md) and
+You do this by wrapping your route with the @scala[`akka.http.scaladsl.server.Route.seal`]@java[`akka.http.javadsl.server.Route.seal`]. The `seal` wrapper applies the logic of the in-scope @ref[ExceptionHandler](exception-handling.md) and
 @ref[RejectionHandler](rejections.md#the-rejectionhandler) to all exceptions and rejections coming back from the route,
 and translates them to the respective `HttpResponse`.
 
-Note that explicit call on the `akka.http.scaladsl.server.Route.seal` method is needed in test code, but in your application code it is not necessary.
+Note that explicit call on the @scala[`akka.http.scaladsl.server.Route.seal`]@java[`akka.http.javadsl.server.Route.seal`] method is needed in test code, but in your application code it is not necessary.
```

Comment:
I suggest we simplify this to `Note that explicit call on the `Route.seal()` method ...`. Most other places do not use the fully qualified name when referring to methods.